### PR TITLE
stops2mimir: Reject stop with invalid coordinate.

### PR DESCRIPTION
**Pull request by kadhikari**
Concerned Ticket: https://jira.canaltp.fr/browse/NAVITIAII-2266

### CODE REVIEW

File: src/bin/stops2mimir.rs [hunk 1]
```diff
@@ -67,6 +72,13 @@ struct Args {
     city_level: u32,
 }
 
+#[derive(Deserialize, Debug)]
+enum StopConvErr {
+    InvisibleStop,
+    StopPoint,
```
Comment (pbougue) [inline src/bin/stops2mimir.rs]: What is the meaning of that error? `NotStopArea`, `StopPointProvided`?
(and not sure about the name of the struct, maybe StopAreaConversionError would be clearer, or something with a "namespace"?)

File: src/bin/stops2mimir.rs [hunk 2]
```diff
@@ -91,9 +103,16 @@ impl GtfsStop {
...
     }
     // to be moved when TryInto is stablilized
-    fn try_into(self) -> Option<mimir::Stop> {
-        if self.location_type == Some(1) && self.visible != Some(0) {
-            Some(mimir::Stop {
+    fn try_into(self) -> Result<mimir::Stop, StopConvErr> {
+        if self.location_type != Some(1) {
+            Err(StopConvErr::StopPoint)
+        } else if self.visible == Some(0){
+        	Err(StopConvErr::InvisibleStop)
```
Comment (pbougue) [inline src/bin/stops2mimir.rs]: looks like it needs a cargo fmt (on all levels you modified, cf .travis.yml file)

File: src/bin/stops2mimir.rs [hunk 3]
```diff
@@ -91,9 +103,16 @@ impl GtfsStop {
...
+        } else if self.stop_lat <= MIN_LAT || self.stop_lat >= MAX_LAT || self.stop_lon <= MIN_LON || self.stop_lon >= MAX_LON {
+         	//Here we return an error message          
+		  	Err(StopConvErr::InvalidStop(format!("Invalid lon or lat for stop {}", self.stop_name).into()))
```
Comment (pbougue) [inline src/bin/stops2mimir.rs]: would be nice to return lat and lon also :)

File: src/bin/stops2mimir.rs [hunk 4]
```diff
@@ -67,6 +72,13 @@ struct Args {
     city_level: u32,
 }
 
+#[derive(Deserialize, Debug)]
+enum StopConvErr {
+    InvisibleStop,
```
Comment (pbougue) [inline src/bin/stops2mimir.rs]: may need a bit more of functional explanation (or where it's used)

File: src/bin/stops2mimir.rs [hunk 5]
```diff
@@ -269,7 +286,14 @@ fn main() {
         })
         .filter_map(|stop: GtfsStop| {
             stop.incr_stop_point(&mut nb_stop_points);
-            stop.try_into()
+            match stop.try_into() {
+                Ok(s) => Some(s),
+                Err(StopConvErr::InvisibleStop) | Err(StopConvErr::StopPoint) => None,
+                Err(StopConvErr::InvalidStop(msg)) => {
+                    warn!("skip csv line because: {}", msg);
```
Comment (pbougue) [inline src/bin/stops2mimir.rs]: minor minor: "because" may be to much

**Comment by kadhikari** [comment]
@pbougue i pushed some correcions

**Comment by kadhikari** [comment]
test this please

File: src/bin/stops2mimir.rs [hunk 6]
```diff
@@ -73,9 +73,12 @@ struct Args {
...
 
 #[derive(Deserialize, Debug)]
-enum StopConvErr {
+enum StopConversionErr {
+    ///StopArea is invisible in Autocomplete
     InvisibleStop,
-    StopPoint,
+    ///The stop in the line is not a StopArea
+    NotStopArea,
+    ///Values of one or more attributs are not valid
```
Comment (pbougue) [inline src/bin/stops2mimir.rs]: typo: attribut`e`s

File: src/bin/stops2mimir.rs [hunk 7]
```diff
@@ -165,11 +171,9 @@ fn attach_stops_to_admins<'a, It: Iterator<Item = &'a mut mimir::Stop>>(
         attach_stop(&mut stop, admins, city_level);
     }
 
-    info!(
-        "there is {}/{} stops without any admin",
-        nb_unmatched,
-        nb_matched + nb_unmatched
-    );
+    info!("there is {}/{} stops without any admin",
```
Comment (pbougue) [inline src/bin/stops2mimir.rs]: minor: there `are`

File: src/bin/stops2mimir.rs [hunk 8]
```diff
@@ -91,9 +106,26 @@ impl GtfsStop {
...
+        {
+            //Here we return an error message
+            Err(StopConversionErr::InvalidStop(
+                format!(
+                    "Invalid lon {:?} or lat {:?} for stop \
+                                                        {:?}",
+                    self.stop_lon,
+                    self.stop_lat,
+                    self.stop_name
+                ).into(),
```
Comment (TeXitoi) [inline src/bin/stops2mimir.rs]: `.into()` is useless here.

File: src/bin/stops2mimir.rs [hunk 9]
```diff
@@ -296,7 +332,15 @@ fn test_load_stops() {
...
-            stop.try_into()
+            match stop.try_into() {
+                Ok(s) => Some(s),
+                Err(StopConversionErr::InvisibleStop) => None,
+                Err(StopConversionErr::NotStopArea) => None,
+                Err(StopConversionErr::InvalidStop(msg)) => {
+                    warn!("skip csv line: {}", msg);
+                    None
+                }
+            }
```
Comment (TeXitoi) [inline src/bin/stops2mimir.rs]: Better to mutualize this copy/pasted code.

**Pull request merged**